File: bat/Format.py
#!/bin/python
##----------------------------------------------------------------------------
##
##       Copyright (C) 2016-2019 Frank Eskesen.
##
##       This file is free content, distributed under the MIT license.
##       (See accompanying file LICENSE.MIT or the original contained
##       within https://opensource.org/licenses/MIT)
##
##----------------------------------------------------------------------------
##
## Title-
##       Format.py
##
## Purpose-
##       Use python formatting to display float value(s).
##
## Last change date-
##       2019/10/26
##
## Usage-
##       Format.py "format string" value ...
##
## Implementation note-
##       See ./test/curltimer for sample usage.
##
##############################################################################
if __name__ == '__main__':
    import sys
    if len(sys.argv) < 3:
        print(sys.argv[0], "requires at least 2 arguments: string value ...")
        sys.exit()

    inps = list(sys.argv[2:])
    outs = []
    for inp in inps:
        try:
            f = float(inp)
            outs += [f]
        except:
            outs += [inp]

    print(str(sys.argv[1]) % tuple(outs))

File: src/cpp/Stock/Util/Format.cpp
//----------------------------------------------------------------------------
//
//       Copyright (c) 2007 Frank Eskesen.
//
//       This file is free content, distributed under the GNU General
//       Public License, version 3.0.
//       (See accompanying file LICENSE.GPL-3.0 or the original
//       contained within https://www.gnu.org/licenses/gpl-3.0.en.html)
//
// SPDX-License-Identifier: GPL-3.0-only
//----------------------------------------------------------------------------
//
// Title-
//       Format.cpp
//
// Purpose-
//       Format the stock data.
//
// Last change date-
//       2007/01/01
//
//----------------------------------------------------------------------------
#include <cstdio>
#include <cstdlib>
#include <cstring>

#include <com/Debug.h>
#include <com/define.h>
#include <com/Reader.h>
#include <com/Writer.h>

//----------------------------------------------------------------------------
// Constants for parameterization
//----------------------------------------------------------------------------
#define __SOURCE__       "FORMAT  " // Source file, for debugging

//----------------------------------------------------------------------------
//
// Subroutine-
//       main
//
// Purpose-
//       Mainline code.
//
//----------------------------------------------------------------------------
int                                 // Return coe
   main(                            // Mainline code
     int             argc,          // Argument count
     char*           argv[])        // Argument array
{
   Reader            reader("composite.inp");
   Writer            writer("composite.out");

   int               c;
   int               rc;

   //-------------------------------------------------------------------------
   // Remove extra line delimiters
   //-------------------------------------------------------------------------
   for(;;)
   {
     c= reader.get();

     if( c == '\n' || c == '\r' )
     {
       while( c == '\n' || c == '\r' )
         c= reader.get();

       writer.put('\r');
       writer.put('\n');
     }

     if( c == EOF )
       break;

     writer.put(c);
   }

   return 0;
}

